File: soln-linkedlist-problems/swap-nodes-in-pairs.java
/*
https://leetcode.com/problems/swap-nodes-in-pairs/
*/

import java.util.*;
import java.lang.*;
import java.io.*;

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head==null || head.next==null)
            return head;
    
        ListNode p=head;
        ListNode q=p.next;

        while(p!=null && q!=null){
            int temp=p.val;
            p.val=q.val;
            q.val=temp;
            
            p=q.next;
            if(p!=null)
                q=p.next;
        }
        return head;
    }
}

File: Python/soln-linkedlist-problems/swap-nodes-in-pairs.py
'''
 @author Farheen Bano
  
 Reference-
 https://leetcode.com/problems/swap-nodes-in-pairs/
'''

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if(head is None or head.next is None):
            return head
    
        p=head
        q=p.next

        while(p is not None and q is not None):
            temp=p.val
            p.val=q.val
            q.val=temp
            
            p=q.next
            if(p is not None):
                q=p.next
        
        return head